firedrake/mesh.py

```diff
@@ -1151,7 +1151,7 @@ def _c_locator(self, tolerance=None):
     int locator(struct Function *f, double *x)
     {
         struct ReferenceCoords reference_coords;
-        return locate_cell(f, x, %(geometric_dimension)d, &to_reference_coords, &reference_coords);
+        return locate_cell(f, x, %(geometric_dimension)d, &to_reference_coords, &to_reference_coords_xtr, &reference_coords);
```

Comment:
It remains unclear to me why one needs a separate function for mapping to reference coordinates on extruded meshes (especially given that it was not needed beforehand).